### PR TITLE
`exp remove`: add `--rev` and `-n` options

### PR DESCRIPTION
**Pull request by karajan1001**
Related to https://github.com/iterative/dvc/pull/7275

> & https://github.com/iterative/dvc/pull/7245

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Same Q: Is it possible to re-do this PR from a branch directly on this repo? Thanks

File: content/docs/command-reference/exp/remove.md [hunk 1]
```diff
@@ -26,7 +27,12 @@ With `--queue`, the list of experiments awaiting execution is cleared instead.
 - `--queue` - remove all experiments that haven't been run yet (defined via
   `dvc exp run --queue`).
 
-- `-A`, `--all` - remove all experiments (includes `--queue`).
+- `-A`, `--all-commits` - Remove all experiments in the repository (overrides
+  `--rev`).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/remove.md:31]: Same as https://github.com/iterative/dvc.org/pull/3192#pullrequestreview-864364698

File: content/docs/command-reference/exp/remove.md [hunk 2]
```diff
@@ -26,7 +27,12 @@ With `--queue`, the list of experiments awaiting execution is cleared instead.
...
+
+- `--rev <commit>` - Remove experiments derived from the specified revision.
+
+- `-n <num>`, `--num <num>` - Remove the last `num` commits from `<commit>`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/remove.md:35]: Same as https://github.com/iterative/dvc.org/pull/3187#pullrequestreview-864359473

**Review by jorgeorpinel — COMMENTED** [review]
Never mind on re-opening the PR, I'm reviewing this as-is... Let's just try to remember for future PRs 🙂 Thanks @karajan1001

File: content/docs/command-reference/exp/show.md [hunk 3]
```diff
@@ -74,7 +74,8 @@ will be generated using the same data from the table.
 - `-A`, `--all-commits` - include experiments derived from all Git commits, as
   well as from the last one. This prints all experiments in the project.
 
-- `-n <num>`, `--num <num>` - show the last `num` commits from HEAD.
+- `-n <num>`, `--num <num>` - show experiments from the `<rev>` commit and `num`
+  commits before it. (`-1` for the whole history of the `<rev>`)
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/show.md:78]: Out of scope? This is being done in https://github.com/iterative/dvc.org/pull/3187 (please revert).

**Comment by jorgeorpinel** [comment]
> @karajan1001 please note I only assigned you for visibility in https://github.com/orgs/iterative/projects/255. Thanks

File: content/docs/command-reference/exp/remove.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/remove.md]: Will need update again per https://github.com/iterative/dvc.org/pull/3187#pullrequestreview-883803230

File: content/docs/command-reference/exp/remove.md [hunk 4]
```diff
@@ -29,6 +30,13 @@ With `--queue`, the list of experiments awaiting execution is cleared instead.
 - `-A`, `--all` - remove all experiments that have been run. Use `--queue` to
   remove queued ones.
 
+- `--rev <commit>` - remove experiments derived from the specified `<commit>` as
+  baseline.
+
+- `-n <num>`, `--num <num>` - show experiments from the previous `num` commits
+  (before the `--rev` baseline). For example `-n -2` includes experiments from
+  `HEAD` and its 2 first ancestors.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/remove.md:38]: See https://github.com/iterative/dvc.org/pull/3187#pullrequestreview-902553693

File: content/docs/command-reference/exp/remove.md [hunk 5]
```diff
@@ -29,6 +30,13 @@ With `--queue`, the list of experiments awaiting execution is cleared instead.
 - `-A`, `--all` - remove all experiments that have been run. Use `--queue` to
   remove queued ones.
 
+- `--rev <commit>` - remove experiments derived from the specified `<commit>` as
+  baseline.
+
+- `-n <num>`, `--num <num>` - show experiments from the `--rev` baseline and
+  from `num` commits before it (first parents). Give a negative value to include
+  all first-parent commits (similar to `git log -n`).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/remove.md:38]: Same as https://github.com/iterative/dvc.org/pull/3187#pullrequestreview-903319359, https://github.com/iterative/dvc.org/pull/3191#pullrequestreview-903953082, https://github.com/iterative/dvc.org/pull/3192#pullrequestreview-903954922
```suggestion
- `-n <num>`, `--num <num>` - show experiments from the last `num` commits
  (first parents) starting from the `--rev` baseline. Give a negative value to include
  all first-parent commits (similar to `git log -n`).
```

**Comment by julieg18** [comment]
Updating this branch with the latest version of master should fix the broken build!

**Review by jorgeorpinel — APPROVED** [review]

**Comment by jorgeorpinel** [comment]
Has the core feature been released or is close to? If so please merge this or lmk and I will. Thanks

**Comment by dberenbaum** [comment]
It will be released in 2.9.6, which will likely happen in the next week or so.

**Comment by jorgeorpinel** [comment]
Well we merged all the others already so... 🚀

**Pull request merged**